pkg/storage/internalstorage/storage.go

```diff
@@ -67,12 +127,25 @@ func (f *StorageFactory) GetResourceVersions(ctx context.Context, cluster string
 }
 
 func (f *StorageFactory) CleanCluster(ctx context.Context, cluster string) error {
-	result := f.db.WithContext(ctx).Where(map[string]interface{}{"cluster": cluster}).Delete(&Resource{})
-	return InterpretDBError(cluster, result.Error)
+	mutex.Lock()
+	tables, err := f.db.Migrator().GetTables()
+	if err != nil {
+		mutex.Unlock()
+		return err
+	}
+	mutex.Unlock()
+
+	for _, table := range tables {
+		result := f.db.WithContext(ctx).Table(table).Where(map[string]interface{}{"cluster": cluster}).Delete(&Resource{})
+		if result.Error != nil {
+			return InterpretDBError(cluster, result.Error)
+		}
+	}
```

Comment:
It is recommended to use transaction deletion

Reply:
done.